### PR TITLE
AC-404:Vitals History Chart

### PR DESCRIPTION
**Pull request by DefCon-007**
I have added the charts to in the patient dashboard. Below are the screenshots.
screenshot_1498799987
screenshot_1498799989
screenshot_1498799979

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        if (chart.getVisibleXRange() > 30 * 6) {
+
+            return monthName + " " + yearName;
+        } else {
+
+            int dayOfMonth = determineDayOfMonth(days, month + 12 * (year - 2016));
+
+            String appendix = "th";
+
+            switch (dayOfMonth) {
```
Comment (AvijitGhosh82) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java]: Nice work! Brownie points for thoughtful UX.

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        int month = determineMonth(days);
+        String monthName = mMonths[month % mMonths.length];
+        String yearName = String.valueOf(year);
+
+        if (chart.getVisibleXRange() > 30 * 6) {
+
+            return monthName + " " + yearName;
+        } else {
+
+            int dayOfMonth = determineDayOfMonth(days, month + 12 * (year - 2016));
```
Comment (AvijitGhosh82) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java]: Why is 2016 hardcoded?

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+            int year = determineYear(daysForMonths);
+            daysForMonths += getDaysForMonth(count % 12, year);
+            count++;
+        }
+
+        return days - daysForMonths;
+    }
+
+    private int determineYear(int days) {
```
Comment (AvijitGhosh82) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java]: Can you generalize this? What if there are old records from 2014?

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java (hunk 1 above)
Comment (DefCon-007) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java]: Thanks :)

File: openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java (hunk 3 above)
Comment (AvijitGhosh82) [inline openmrs-client/src/main/java/org/openmrs/mobile/utilities/DayAxisValueFormatter.java]: @DefCon-007

**Pull request merged**